src/Integration/MefServices/ActiveSolutionBoundTracker.cs

```diff
@@ -102,17 +108,28 @@ public ActiveSolutionBoundTracker(IHost host, IActiveSolutionTracker activeSolut
 
             if (isSolutionCurrentlyBound)
             {
-                var solutionBinding = this.extensionHost.GetService<ISolutionBindingSerializer>();
-                solutionBinding.AssertLocalServiceIsNotNull();
-                BoundSonarQubeProject boundProject = solutionBinding.ReadSolutionBinding();
-                var connectionInformation = boundProject.CreateConnectionInformation();
-
-                // TODO: handle connection failure
-                // TODO: block until the connection is complete
-                await this.extensionHost.SonarQubeService.ConnectAsync(connectionInformation, CancellationToken.None);
+                try
+                {
+                    var connectionInformation = GetConnectionInformation();
+                    await sonarQubeService.ConnectAsync(connectionInformation, CancellationToken.None);
+                }
+                catch (Exception ex)
+                {
+                    sonarLintOutput.Write($"Cannot connect to SonarQube: '{ex.Message}'");
```

Comment:
Out of interest, does the output window message add timestamp? It might be useful to get an idea of the rate at which errors are occurring.

Reply:
It currently doesn't.